src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.V1.PayloadProduction.cs

```diff
@@ -340,6 +340,63 @@ public async Task getPayloadV1_picks_transactions_from_pool_constantly_improving
         txs.Should().HaveCount(11);
     }
 
+    [Test, Repeat(100)]
+    public async Task Cannot_produce_bad_blocks()
```

Comment:
reproduced invalid block

Reply:
Magic! All of the surrounding logic has treated `.ForceProcessing` as if it does not reuse that list of blocks (see this [lines](https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Nethermind.Consensus/Processing/BlockchainProcessor.cs#L502-L506)...  Wonder if it will be able also to check if a single iteration exit on the loop end (something like [this](https://github.com/NethermindEth/nethermind/commit/45a589d5e0b9d05ac5a5736ef3c418f69849ddc3#diff-d3815f34e7dae0c4c0af79174d6c0d34c4c9d5c9e7259c23a56379cea9f72d51R621)) is also only suitable for the `.Trace` case and not entire `.ForceProcessing`)